File: algorithm/swift/fibonacci.swift
import Foundation

func fib(n: Int) -> Int {
    if n < 2 {
        return n
    }
    return fib(n: n - 2) + fib(n: n - 1)
}

let start = Date()

print("result: \(fib(n: 40))")

let elapsed = Date().timeIntervalSince(start)

print("time: \(elapsed) sec")

File: algorithm/ruby/fibonacci.rb
require 'benchmark'

def fib(n)
   if (n < 2) then
       return n
   end
   return fib(n - 2) + fib(n - 1)
end

result = Benchmark.realtime do
  puts "result: #{fib(40)}"
end

puts "time: #{result} sec"